File: src/main/java/dev/max/invana/entities/AgentSettings.java
package dev.max.invana.entities;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class AgentSettings {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String token;
    private boolean tlsRequired;
    private int collectionInterval;
    private int retryAttempts;
    private int timeout;
    private boolean detailedLogging;
    private String serverUrl;
    private int serverPort;

    @Lob
    private String customHeadersJson;

}

File: frontend/src/types/AgentSettings.ts
export type AgentSettings = {
    id: number;
    token: string;
    tlsRequired: boolean;
    collectionInterval: number;
    retryAttempts: number;
    timeout: number;
    detailedLogging: boolean;
    serverUrl: string;
    serverPort: number;
    customHeadersJson: string;
  };
  